Kernel/Tasks/Thread.h

```diff
@@ -94,11 +95,12 @@ class Thread
     Process& process() { return m_process; }
     Process const& process() const { return m_process; }
 
-    SpinlockProtected<NonnullOwnPtr<KString>, LockRank::None> const& name() const
+    SpinlockProtected<NonnullRefPtr<TaskName>, LockRank::None> const& name() const
     {
         return m_name;
     }
-    void set_name(NonnullOwnPtr<KString> name);
+    ErrorOr<void> set_name(NonnullOwnPtr<KString> name);
```

Comment:
We might want to put a bold difference this and the other method... something like `set_new_name` could be OK?